src/chains/ethereum/ethereum/tests/provider.test.ts

```diff
@@ -30,6 +30,154 @@ describe("provider", () => {
+      assert(
+        parseInt(block.timestamp) >= sentAfter,
+        `block wasn't mined at the right time, should have been on or after ${sentAfter}, was ${parseInt(
+          block.timestamp
+        )}`
+      );
+      assert(
+        parseInt(block.timestamp) <= sentBefore,
+        `block wasn't mined at the right time, should have been on or before ${sentAfter}, was ${parseInt(
+          block.timestamp
+        )}`
+      );
+      await provider.disconnect();
+    });
+
+    it("uses the timestampIncrement option", async () => {
+      const time = new Date("2019-01-01T00:00:00.000Z");
+      const timestampIncrement = 5;
+      const provider = await getProvider({
+        chain: { time },
+        miner: { timestampIncrement }
+      });
+      const accounts = await provider.send("eth_accounts");
+      const tx = { from: accounts[0], gas: "0xfffff" };
+      const hash = await provider.request({
+        method: "eth_sendTransaction",
+        params: [tx]
+      });
+      const receipt = await provider.request({
+        method: "eth_getTransactionReceipt",
+        params: [hash]
+      });
+      assert.notStrictEqual(receipt, null);
+      const block = await provider.request({
+        method: "eth_getBlockByNumber",
+        params: ["latest", false]
+      });
+      assert.strictEqual(
+        parseInt(block.timestamp),
+        +time / 1000 + timestampIncrement
+      );
+      await provider.disconnect();
+    });
+
+    it("uses the timestampIncrement for the first block when forking", async () => {
+      const time = new Date("2019-01-01T00:00:00.000Z");
+      const timestampIncrement = 5;
+      const fakeMainnet = await getProvider({
+        chain: { time }
+      });
+      const provider = await getProvider({
+        fork: { provider: fakeMainnet as any },
+        miner: { timestampIncrement }
+      });
+      const block = await provider.request({
+        method: "eth_getBlockByNumber",
+        params: ["latest", false]
+      });
+      assert.strictEqual(
+        parseInt(block.timestamp),
+        +time / 1000 + timestampIncrement
+      );
+      await provider.disconnect();
+      await fakeMainnet.disconnect();
+    });
+    it("uses the `time` option for the first block even when when `timestampIncrement` is not 'clock' when forking", async () => {
+      const time = new Date("2019-01-01T00:00:00.000Z");
+      const timestampIncrement = 5;
+      const fakeMainnet = await getProvider({
+        chain: { time }
+      });
+      const time2 = new Date("2020-01-01T00:00:00.000Z");
+      const provider = await getProvider({
+        fork: { provider: fakeMainnet as any },
+        chain: { time: time2 },
+        miner: { timestampIncrement }
+      });
+      const block = await provider.request({
+        method: "eth_getBlockByNumber",
+        params: ["latest", false]
+      });
+      assert.strictEqual(parseInt(block.timestamp), +time2 / 1000);
+      await provider.disconnect();
```

Comment:
What's the implication of _not_ disconnecting here?

I see that we don't disconnect any of the other providers in this file.

Reply:
We really should always disconnect, but for forking it's more important due to the persistent cache being unable to be shared between instances.